velox/exec/MergeJoin.cpp

```diff
@@ -273,11 +290,18 @@ void MergeJoin::prepareOutput() {
     outputSize_ = 0;
 
     if (filterInput_ != nullptr) {
-      // When filterInput_ contains array or map columns, their child vectors
-      // (elements, keys and values) keep growing after each call to
-      // 'copyRow'. Call BaseVector::resize(0) on these child vectors to avoid
-      // that.
-      for (auto& child : filterInput_->children()) {
+      for (auto i = 0; i < filterInputType_->size(); ++i) {
+        auto& child = filterInput_->childAt(i);
+        if (filterInputToOutputChannel_.find(i) !=
+            filterInputToOutputChannel_.end()) {
+          child = output_->childAt(filterInputToOutputChannel_[i]);
+          continue;
+        }
+
+        // When filterInput_ contains array or map columns, their child vectors
+        // (elements, keys and values) keep growing after each call to
+        // 'copyRow'. Call BaseVector::resize(0) on these child vectors to avoid
```

Comment:
This comment seems outdated.

Reply:
> When filterInput_ contains array or map columns, their child vectors
>        // (elements, keys and values) keep growing after each call to
>       // 'copyRow'.

Would you double check that we do not have this problem with the current code?

Reply:
This is still relevant when the complex typed column in filter input is not projected to output.